[Documentation] Add join metric example for running pods requesting memory (#696)
kubernetes/kube-state-metrics: docs/README.md

- [Pod Metrics](pod-metrics.md)
- [Pod Disruption Budget Metrics](poddisruptionbudget-metrics.md)
- [ReplicaSet Metrics](replicaset-metrics.md)
- [ReplicationController Metrics](replicationcontroller-metrics.md)
- [ResourceQuota Metrics](resourcequota-metrics.md)
- [Service Metrics](service-metrics.md)
- [StatefulSet Metrics](statefulset-metrics.md)
- [Namespace Metrics](namespace-metrics.md)
- [Horizontal Pod Autoscaler Metrics](horizontalpodautoscaler-metrics.md)
- [Endpoint Metrics](endpoint-metrics.md)
- [Secret Metrics](secret-metrics.md)
- [ConfigMap Metrics](configmap-metrics.md)
- [Ingress Metrics](ingress-metrics.md)

## Join Metrics

When an additional, not provided by default label is needed, a [Prometheus matching operator](https://prometheus.io/docs/prometheus/latest/querying/operators/#vector-matching)
can be used to extend single metrics output.

This example adds `label_release` to the set of default labels of the `kube_pod_status_ready` metric
and allows you select or group the metrics by helm release label:

```
kube_pod_status_ready * on (namespace, pod) group_left(label_release)  kube_pod_labels
```

> Another useful example could be to know the amount of memory resources in use by pods in a specific `phase`, for example `Running`:

Reviewer: ```suggestion
Another useful example would be to query the memory usage of pods by its `phase`, like `Running`:
```

```

> sum(kube_pod_container_resource_requests_memory_bytes) by (namespace, pod, node)

Reviewer 2: What does `node` add here?  Each pod can only be on one node.
